core/src/org/samtools/htsjdk/core/Locatable.java

```diff
@@ -0,0 +1,140 @@
+ * </p>
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ * @implSpec note that 0-length intervals are allowed in an implementation-dependent manner.
+ */
+public interface Locatable {
+
+    /**
+     * Represents the reference name for unplaced locations.
+     *
+     * @see #getReferenceName()
+     */
+    public static final String UNPLACED_REFERENCE_NAME = "*";
+
+    /**
+     * Gets the reference name where this location is placed.
+     *
+     * <p>
+     * {@link Locatable} should either be placed on a reference sequence or being unplaced
+     * ({@link #UNPLACED_REFERENCE_NAME}).
+     * </p>
+     *
+     * @return name of the reference for the location; {@link #UNPLACED_REFERENCE_NAME} if unplaced.
+     * Never {@code null}
+     */
+    String getReferenceName();
+
+    /**
+     * Gets the start position on the reference (1-based).
+     *
+     * <p>
+     * {@link Locatable} should have a non-negative location on the reference sequence; if unplaced
+     * ({@code getReferenceName() == UNPLACED_REFERENCE_NAME}), position is undefined.
+     * </p>
+     *
+     * @return 1-based start position on the reference; undefined if {@link #getReferenceName()}
+     * returns {@link #UNPLACED_REFERENCE_NAME}.
+     */
+    long getStart();
+
+    /**
+     * Gets the end position on the reference (1-based fully-closed) if present.
+     *
+     * @return 1-based fully-closed if present; empty optional otherwise.
+     *
+     * @implSpec if present, end position should be downstream te start position ({@code getStart()
+     * > getEnd()}).
+     * @see #getStart()
+     */
+    OptionalLong getEnd();
+
+    /**
+     * Gets the number of bases of reference covered by this location.
+     *
+     * <p>
+     * If end is present, returns the number of bases on te reference from the start to the end
+     * (independently of the alignment, if any); otherwise, it would return an empty value
+     * (unknown).
+     * </p>
+     *
+     * @return number of bases of reference; empty if unknown.
+     *
+     * @implNote default implementation uses {@link #getStart()} and {@link #getEnd()} to compute
+     * the length on the reference.
+     * @see #getStart()
+     * @see #getEnd()
+     */
+    default OptionalLong getLengthOnReference() {
+        final OptionalLong end = getEnd();
+        return end.isPresent() ? OptionalLong.of(end.getAsLong() - getStart() + 1)
+                : OptionalLong.empty();
+    }
+
+    /**
+     * Determines whether this location overlaps with other.
+     *
+     * @param other location to test.
+     *
+     * @return {@code true} if this location overlaps other; {@code false} otherwise.
+     *
+     * @implNote default implementation returns {@code withinDistanceOf(other, 0)}.
+     * @implSpec should return {@code false} if {@code referenceMatch(other) == false}.
+     * @see #withinDistanceOf(Locatable, int)
+     */
+    default boolean overlaps(Locatable other) {

```

Comment:
We can add also more methods to pad/merge/etc. to make the interface complete...